eris_test.go

```diff
@@ -76,61 +109,63 @@ func TestErrorWrapping(t *testing.T) {
 	}
 }
 
-type withMessage struct {
-	msg string
-}
-
-func (e *withMessage) Error() string { return e.msg }
-
-type withLayer struct {
-	err error
-	msg string
-}
-
-func (e *withLayer) Error() string { return e.msg + ": " + e.err.Error() }
-func (e *withLayer) Unwrap() error { return e.err }
-
-type withEmptyLayer struct {
-	err error
-}
-
-func (e *withEmptyLayer) Error() string { return e.err.Error() }
-func (e *withEmptyLayer) Unwrap() error { return e.err }
-
 func TestExternalErrorWrapping(t *testing.T) {
 	tests := map[string]struct {
 		cause  error    // root error
 		input  []string // input for error wrapping
-		output error    // expected output
+		output []string // expected output
```

Comment:
can we not use the output as an error type rather than a []string type?

Reply:
this way is more accurate imo and allows us to test pkg/errors style cases where there are two layers with the same message (that test case wasn't possible previously)